Stole from gentoo bugzilla
http://bugs.gentoo.org/show_bug.cgi?id=156475

Lets this work with gcc 4.2/4.3
-Khem
Index: glib-2.16.1/glib/gutils.h
===================================================================
--- glib-2.16.1.orig/glib/gutils.h	2008-03-10 17:31:56.000000000 -0700
+++ glib-2.16.1/glib/gutils.h	2008-06-04 15:19:12.000000000 -0700
@@ -96,13 +96,14 @@
 #ifdef G_IMPLEMENT_INLINES
 #  define G_INLINE_FUNC
 #  undef  G_CAN_INLINE
-#elif defined (__GNUC__) 
-#  ifdef __GNUC_STDC_INLINE__
-#   define G_INLINE_FUNC extern inline __attribute__ ((__gnu_inline__))
+#elif defined (__GNUC__)
+#  if __GNUC_PREREQ (4,2) && defined (__STDC_VERSION__) \
+   && __STDC_VERSION__ >= 199901L
+#    define G_INLINE_FUNC extern __inline __attribute__ ((__gnu_inline__))
 #  else
-#   define G_INLINE_FUNC extern inline
+#    define G_INLINE_FUNC extern __inline
 #  endif
-#elif defined (G_CAN_INLINE) 
+#elif defined (G_CAN_INLINE)
 #  define G_INLINE_FUNC static inline
 #else /* can't inline */
 #  define G_INLINE_FUNC
